datacube/scripts/dataset.py

```diff
@@ -469,18 +469,29 @@ def _get_derived_set(index: Index, id_: str) -> Set[Dataset]:
 @click.argument('ids', nargs=-1)
 @ui.pass_index()
 def archive_cmd(index: Index, archive_derived: bool, dry_run: bool, ids: List[str]):
-    for id_ in ids:
-        target_dataset = index.datasets.get(id_)
-        if target_dataset is None:
-            echo(f'No dataset found with id {id_}')
-            sys.exit(-1)
+    datasets_for_archive = {dataset_id: exists for dataset_id, exists in zip(ids, index.datasets.bulk_has(ids))}
 
-        to_process = _get_derived_set(index, id_) if archive_derived else [target_dataset]
-        for d in to_process:
-            click.echo('archiving %s %s %s' % (d.type.name, d.id,
-                                               d.local_uri if d.local_uri is not None else "<no location>"))
-        if not dry_run:
-            index.datasets.archive(d.id for d in to_process)
+    if False in datasets_for_archive.values():
+        for dataset_id, exists in datasets_for_archive.items():
+            if not exists:
+                click.echo(f'No dataset found with id: {dataset_id}')
+        sys.exit(0)
```

Comment:
This should not be `0`.

You can use `expect_success=False` argument when using `clirunner`, it won't assert then. You can then check exit code is of expected value in the test.

Reply:
Example of checking for exit code other than 0

https://github.com/opendatacube/datacube-core/blob/afca22831eecf8092e0c293064a9ea465aa6d778/integration_tests/test_validate_ingestion.py#L31-L35

Reply:
Thanks for pointing me in the right direction, I've updated the code :+1: